### PR TITLE
built out a page on the DynamoDB pricing structure 

### PR DESCRIPTION
**Pull request by alecbw**
[x] added `dynamodb-pricing.md`
[x] updated `nav` in `mkdocs.yml`

Generally tried to stay within what I felt was your style, but certainly open to aesthetic (and content) feedback.

I thought about adding in code samples, but felt it wasn't in-style. Your MKDocs build also appears to not have the plugins necessary to use Tabbed Admonitions.

Thoughts @Bensign and co?

### CODE REVIEW

**Comment by Bensign** [comment]
@alecbw Thank you for the effort on this - this is awesome! I pulled this down and went through it and it generally looks good! Let me spend a bit more time tomorrow on this and merge it in over the weekend. 

As for code samples, we aren't opposed to it where it adds value. The closest thing to it right now is an example of ECR lifecycle policies which you can see at the bottom of this page: https://handbook.vantage.sh/aws/services/ecr-pricing/

What were you hoping to use Tabbed Admonitions for?

File: docs/aws/services/dynamodb-pricing.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+|----------------------|----------------------------|
+| Standard single item | 1 WRU                      |
+| Standard batch       | 1 WRU (per item)           |
+| Transactional        | 2 WRU (2x)                 |
+| Oversize 4 KB item   | 4 WRU (4x, size dependent) |
+
+
+## Read Type
+
+Part of what makes DynamoDB a compelling offering is its hybrid approach to the CAP theorem - it can adjust between eventually and strongly Consistent as needed.
```
Comment (Bensign) [inline docs/aws/services/dynamodb-pricing.md:53]: @alecbw Do you mind explaining what the CAP theorem is in a footnote - or linking out from a footnote for an explanation? I'm not sure this is common knowledge. There is an example of footnotes on this page here: http://localhost:8000/aws/services/cloudwatch-pricing/

File: docs/aws/services/dynamodb-pricing.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+Transactions (`client.execute_transaction`) group up to 25 writes (or reads, updates, or deletes) together and *guarantee that they succeed or fail together.*
+
+For a given write of an item up to *1 KB* in size:
+
+| Type                 | Cost                       |
+|----------------------|----------------------------|
+| Standard single item | 1 WRU                      |
+| Standard batch       | 1 WRU (per item)           |
+| Transactional        | 2 WRU (2x)                 |
+| Oversize 4 KB item   | 4 WRU (4x, size dependent) |
```
Comment (Bensign) [inline docs/aws/services/dynamodb-pricing.md:48]: Is this supposed to be `WCU`? I actually can't find a reference to `WRU` anywhere on the pricing page but that doesn't mean I'm right. 😄 

Also for acronyms like this, the first time they're referenced we try to write out the full name and then put the acronym in parenthesis to educate folks as to what it stands for.

File: docs/aws/services/dynamodb-pricing.md (hunk 1 above)
Comment (Bensign) [reply docs/aws/services/dynamodb-pricing.md:53]: Also slight grammar nitpick: do you mind removing the capitalization from `Consistent`?

File: docs/aws/services/dynamodb-pricing.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+That said, some business logic unequivocally dictates strongly consistent reads (for example: an ATM reading a customer's balance).
+
+For a given read of an item up to *4 KB* in size:
+
+| Type                  | Cost       |
+|-----------------------|------------|
+| Eventually Consistent | 0.5 RRU    |
+| Strongly Consistent   | 1 RRU (2x) |
+| Transactional         | 2 RRU (4x) |
```
Comment (Bensign) [inline docs/aws/services/dynamodb-pricing.md:65]: For the first reference to RRU do you mind just spelling it out as `Read Request Unit (RRU)` then use the acronym thereafter to just help educate folks as to what it means?

File: docs/aws/services/dynamodb-pricing.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+Every DynamoDB table has a `partition_key` (`pk`) and optional `sort_key` (`sk`) specified at the time of creation.
+
+Indexes allow alternate partition and sort keys to be used to query items. They may be created at any time and are automatically maintained as new items are written.
+
+Indexes can help control costs in two primary ways:
+
+1. Queries on a new index return less unnecessary items (than the alternative/existing query) and thus cost less RRUs.
+
+2. Each Index optionally allows a subset of item attributes to be **Projected** to that index. Projecting a subset can save on read costs if items are regularly >4 KB, but the projected attribute names+values sum to <4KB
```
Comment (Bensign) [inline docs/aws/services/dynamodb-pricing.md:94]: I believe both "Index" and "Projected" don't need to be capitalized in the first sentence here.

File: docs/aws/services/dynamodb-pricing.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+| Type       | Primary Key Attributes                       |
+|------------|----------------------------------------------|
+| Base table | Initial `pk` + optional `sk`                 |
+| **LSI**    | Initial `pk` + *different* `sk`              |
+| **GSI**    | *Different* `pk` + optional *different* `sk` |
```
Comment (Bensign) [inline docs/aws/services/dynamodb-pricing.md:100]: What do LSI and GSI mean here?

**Review by Bensign — CHANGES_REQUESTED** [review]
Generally looks good! A few questions, clarifications and minor grammar requests.

**Comment by alecbw** [comment]
I realize code samples do add a substantial scope creep to this, so I'll leave it open to your team if that fits the plan.

> What were you hoping to use Tabbed Admonitions for?
to provide code snippets - as an example:
image

File: docs/aws/services/dynamodb-pricing.md (hunk 1 above)
Comment (alecbw) [reply docs/aws/services/dynamodb-pricing.md]: Updated both

File: docs/aws/services/dynamodb-pricing.md (hunk 2 above)
Comment (alecbw) [reply docs/aws/services/dynamodb-pricing.md:55]: The acronym WRU appears to mostly be used by third party vendors and how-to guides. I added a section under BillingMode to clarify the relevant units for each Billing mode type
image

File: docs/aws/services/dynamodb-pricing.md (hunk 4 above)
Comment (alecbw) [reply docs/aws/services/dynamodb-pricing.md]: Updated in the newest commit

File: docs/aws/services/dynamodb-pricing.md (hunk 5 above)
Comment (alecbw) [reply docs/aws/services/dynamodb-pricing.md]: Updated in the newest commit

**Comment by Bensign** [comment]
Ah, let me bring up tabbed admonitions and code snippets up in our sync this week to discuss. Separately this looks good. I'm going to merge it in and deploy momentarily! Thanks again for your contributions!

**Pull request merged**